template/gitlab-runner.tftpl

```diff
@@ -39,17 +39,45 @@ then
     gitlab_runner_registration_token=$(aws ssm get-parameter --name "${secure_parameter_store_gitlab_runner_registration_token_name}" --with-decryption --region "${secure_parameter_store_region}" | jq -r ".Parameter | .Value")
 fi
 
+# fetch gitlab token from SSM
+if [[ "${gitlab_token}" == "__GITLAB_TOKEN_FROM_SSM__" ]]
+then
+    gitlab_token=$(aws ssm get-parameter --name "${secure_parameter_store_gitlab_token_name}" --with-decryption --region "${secure_parameter_store_region}" | jq -r ".Parameter | .Value")
+fi
+
 if [[ "${runners_token}" == "__REPLACED_BY_USER_DATA__" && "$token" == "null" ]] || [[ "$valid_token" == "false" ]]
 then
-  token=$(curl ${curl_cacert} --request POST -L "${runners_gitlab_url}/api/v4/runners" \
-    --form "token=$gitlab_runner_registration_token" \
-    --form "tag_list=${gitlab_runner_tag_list}" \
-    --form "description=${gitlab_runner_description}" \
-    --form "locked=${gitlab_runner_locked_to_project}" \
-    --form "run_untagged=${gitlab_runner_run_untagged}" \
-    --form "maximum_timeout=${gitlab_runner_maximum_timeout}" \
-    --form "access_level=${gitlab_runner_access_level}" \
-    | jq -r .token)
+  if [[ "${gitlab_runner_version}" < "16.0.0" ]]
+  then
+    token=$(curl ${curl_cacert} --request POST -L "${runners_gitlab_url}/api/v4/runners" \
+      --form "token=$gitlab_runner_registration_token" \
+      --form "tag_list=${gitlab_runner_tag_list}" \
+      --form "description=${gitlab_runner_description}" \
+      --form "locked=${gitlab_runner_locked_to_project}" \
+      --form "run_untagged=${gitlab_runner_run_untagged}" \
+      --form "maximum_timeout=${gitlab_runner_maximum_timeout}" \
+      --form "access_level=${gitlab_runner_access_level}" \
+      | jq -r .token)
+  else
+    runner_type_param=""
+    if [ "${gitlab_runner_type}" = "group_type" ]; then
+      runner_type_param='--form group_id=${gitlab_runner_group_id}'
+    elif [ "${gitlab_runner_type}" = "project_type" ]; then
+      runner_type_param='--form project_id=${gitlab_runner_project_id}'
+    fi
+
```

Comment:
suggestion: It might happen, that the module caller forgot to set both `group_id` and `project_id`. Better to throw an error here to make the caller aware of this situation.